File: Interviews/LeetCode/Daily-Problems/2022/November/Day-10/1047.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        string ans = "";
        for (char c: s) {
            if (ans.size() and ans.back() == c) {
                while (ans.size() > 0 and ans.back() == c) {
                    ans.pop_back();
                }
            } else {
                ans.push_back(c);   
            }
        }
        
        return ans;
    }
};

File: Interviews/LeetCode/Daily-Problems/2022/November/Day-10/1047.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        ans = []
        for c in s:
            if len(ans) and ans[-1] == c:
                while ans and ans[-1] == c:
                    ans.pop()
            else:
                ans.append(c)
        
        return "".join(ans)